src/hotspot/share/runtime/handshake.cpp

```diff
@@ -612,14 +612,11 @@ void HandshakeState::do_self_suspend() {
   assert(Thread::current() == _handshakee, "should call from _handshakee");
   assert(_lock.owned_by_self(), "Lock must be held");
   assert(!_handshakee->has_last_Java_frame() || _handshakee->frame_anchor()->walkable(), "should have walkable stack");
-  JavaThreadState jts = _handshakee->thread_state();
+
```

Comment:
Can we add an assert:

`assert(_handshakee->thread_state() == _thread_blocked, "Caller should have transitioned to _thread_blocked);
`
?

Reply:
Done!

Reply:
Should be able to add that assert. The new code has only two callers
to do_self_suspend() and both make sure we are in `_thread_blocked`.